dandiapi/api/views/zarr.py

```diff
@@ -169,3 +174,53 @@ def delete_files(self, request, zarr_id):
             paths = [file['path'] for file in serializer.validated_data]
             zarr_archive.delete_files(paths)
         return Response(None, status=status.HTTP_204_NO_CONTENT)
+
+
+@swagger_auto_schema(
+    method='GET',
+    responses={
+        200: 'foobar',
+    },
+    manual_parameters=[
+        openapi.Parameter(
+            'path',
+            openapi.IN_PATH,
+            'a file or directory path within the zarr file.',
+            type=openapi.TYPE_STRING,
+            required=True,
+            pattern=r'.*',
+        )
+    ],
+)
+@api_view(['GET'])
+def explore_zarr_archive(request, zarr_id: str, path: str):
+    zarr_archive = get_object_or_404(ZarrArchive, zarr_id=zarr_id)
+    if path == '':
+        path = '/'
+    # If a path ends with a /, assume it is a directory.
+    # Return a JSON blob which contains URLs to the contents of the directory.
+    if path.endswith('/'):
+        # Strip off the trailing /, it confuses the ZarrChecksumFileUpdater
+        path = path[:-1]
+        # We use the .checksum file to determine the directory contents, since S3 cannot.
+        listing = ZarrChecksumFileUpdater(
+            zarr_archive=zarr_archive, zarr_directory_path=path
+        ).read_checksum_file()
+        if listing is None:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+        directories = [
+            settings.DANDI_API_URL + reverse('zarr-explore', args=[zarr_id, directory.path])
+            for directory in listing.checksums.directories
+        ]
+        files = [
+            settings.DANDI_API_URL + reverse('zarr-explore', args=[zarr_id, file.path])
+            for file in listing.checksums.files
+        ]
+        return Response({'directories': directories, 'files': files})
```

Comment:
The `asset_paths` endpoint returns directory structures in the format `{ 'folders': [...], 'files': [...] }`. I think we should decide on using `folders` or `directories` to keep things consistent.

Reply:
I called it `directories` all over the `zarr_checksum` model and in the `.checksum` folder, so I vote we switch `asset_paths` to use `directories` as well. IIRC "folders" are a windows terminology anyway.

Reply:
Hear hear! I've never liked the creeping usage of "folder" for "directory", and I've been staking out my position on UNIX hill for years now.